docs/solutions/k8s-on-dcos.md

```diff
@@ -0,0 +1,271 @@
+* Applications that run on-premise, in a private datacenter.
+* The need for a multi-service setup, on the same cluster, is a given.
+* A requirement to easily migrate parts or all of the distributed application from a private cloud to a public cloud setup, including Amazon Web Services, Google Compute and Microsoft Azure.
+* Run apps in a highly-automated environment in order to reduce operational burden.
+* Reliability and fault-tolerance for all workloads.
+* Enterprise-grade security.
+* Elastic scaling of resources between Kubernetes and other apps.
+
+In the following we will have a look at two real-world case studies in greater detail.
+
+### 2.1 Case study 1: Financial Industry
+
+A leading, multi-national investment bank has a number of applications—from fraud detection to trading and marketing—running on the Mesosphere DCOS and decides to leverage Kubernetes on DCOS in order to better manage their workloads. The majority of the datasets reside on-premise due to various legislative requirements (Sarbanes-Oxley, Dodd-Frank, etc.). However, especially for targeted marketing and sentiment analysis use cases, said investment bank has a 500 node cluster in a public cloud, in addition to their on-premise machines. Kubernetes on DCOS in this scenario is used to enable them to build and run multiple services—stream processing of public news and social media feeds, machine learning to detect fraudulent behavior, recommendations for customers and reaching out to prospects—on both  their private and public cloud machines.
+
+### 2.2 Case study 2: Gaming
+
+In this case study, we look at a Web 2.0 property which has its main line of business in online games deployed through Web sites and mobile applications. The two main challenges the gaming company faced was to ensure a high utilization of their on-premise cluster and being flexible concerning deployment options. Essentially, they wanted to have game engines built in Kubernetes and also run analytics (Spark, Hadoop, Storm) together on the same cluster in order to lower their TCO (Total Cost of Ownership) and increase their RoI (Return on Investment). As they rolled out the system, Kubernetes on DCOS helped them to create a reliable and flexible architecture, allowing them to triple their compute utilization.
+
+## 3. Solution
+
+Let us now have a look at how Kubernetes is integrated with the Mesosphere DCOS from a 50,000 ft. view:
+
+![Kubernetes on DCOS architecture overview](k8s-on-dcos/k8s-on-dcos_arch_overview.png)
+
+Kubernetes on DCOS begins with a Mesosphere DCOS cluster built atop bare metal or virtual machines running any modern version of Linux. The nodes can either be on-premise or in a public cloud/IaaS setting, such as Google Compute Platform. The Mesosphere DCOS is built around Apache Mesos forms the distributed systems kernel that abstracts away all of the underlying nodes so that they combine to form what looks like one big machine.
```

Comment:
@mhausenblas @davidopp there are a couple of [known issues][1] related to networking.  TL;DR: we don't use flannel, we use a custom endpoints service controller that maps services to HostIP:HostPort instead of PodIP:PodPort. We do this because we want to assume that Docker is running in its default networking mode on each slave host, without any custom IP configuration.

The (biggest) ugly consequence of this approach is that right now, for each container Port that will become a backing endpoint for a service, users are required to manually specify a non-zero HostPort that falls into the range of the "ports" resources declared on a slave: it forces users to track their own usage of the host's port namespace! There is work in progress to relieve the user of that burden but it's not committed to master yet.

[1]: https://github.com/mesosphere/kubernetes-mesos/blob/master/docs/issues.md